pyctcdecode/language_model.py

```diff
@@ -362,9 +363,16 @@ def parse_directory_contents(filepath: str) -> Dict[str, str]:
         }
 
     @classmethod
-    def load_from_dir(cls, filepath: str) -> "LanguageModel":
+    def load_from_dir(cls, filepath: Optional[str] = None, filenames: Optional[Dict] = None) -> "LanguageModel":
```

Comment:
would this be ok? Every single file is downloaded and hashed individually from the hub and stored in a cache -> it's not really possible to cache a whole dir

Reply:
Looking for additional thoughts here: 
I'd suggest this get split out into two methods: `load_from_dir` and `load_from_filenames`

`load_from_dir` can basically just verify directory contents and pass filenames to `load__from_filenames`, which will do most of the work that's done now.

this way there's a bit more encapsulation of functionality (it'll also be slightly easier to test)

Reply:
That sounds good to me! Will split the function into two